bundles/core/src/main/java/com/adobe/cq/wcm/core/components/commons/link/Link.java

```diff
@@ -74,13 +72,24 @@ default String getURL() {
     }
 
     /**
-     * The processed URL, in case any extra processing can be done.
+     * The mapped URL, which supports mapping and vanity path.
      * This usually is resource resolver mapping.
      *
-     * @return Process link URL or {@code null} if link is invalid or no processing can be done
+     * @return Mapped link URL or {@code null} if link is invalid or no processing can be done
      */
     @Nullable
-    default String getProcessedURL() {
+    default String getMappedURL() {
+        return null;
+    }
+
+
+    /**
+     * The externalized URL which also contains the scheme and host information.
```

Comment:
I'd go with: Returns the externalized URL ...

Reply:
The @return tag should make this clear. It's also not aligned with other interfaces like OEmbedClient for example.